src/collision/cm_trace.c

```diff
@@ -246,91 +246,168 @@ static void Cm_TestInLeaf(cm_trace_data_t *data, int32_t leaf_num) {
 }
 
 /**
+ * @brief Temporary structured used for a single trace node
```

Comment:
Dude, this `cm_*` stuff really has to be done in another branch. We can't risk regression like this in trying to get stain maps in before release. Do you agree?